config.go

```diff
@@ -325,26 +325,28 @@ type Config struct {
 	// loadConfig function. We need to expose the 'raw' strings so the
```

Comment:
this commit is doing way too much. 
- start with just `lnwire` (ie, "schema first" rule) and make sure to add the necessary tests around decoding/encoding there. Also, maybe i missed this but somewhere we need to check that our peers are not sending us more than 1 DNS address in their announcements (see bolt 7). 
- then slowly expand out from there, commit by commit to start using the new type. 

each commit should be one unit of change that the reviewer needs to consider.